.github/workflows/mcad-CI.yml

```
@@ -0,0 +1,36 @@
+# This workflow will build mcad and run unit test and E2E tests
+name: MCAD-CI
+on:
+  pull_request:
+    branches:
+      - 'main'
```

```suggestion
      - 'main'
      - 'release-*'
```
To cover possible patch branches.